middle_end/flambda/simplify/env/rec_uses.mli

```diff
@@ -0,0 +1,83 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*    Pierre Chambart and Guillaume Bury, OCamlPro                        *)
+(*                                                                        *)
+(*   Copyright 2021--2021 OCamlPro SAS                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** Rec uses
```

Comment:
A posteriori, I'm not convinced by the name of this module. It is used to remove unused arguments of recursive continuations. But the uses themselves are not really recursive. Well, I don't really have a good name to propose either.
`Variable_dependency_graph` (maybe not, we might end up using that for symbols also), `Uses` ?

Reply:
`Uses` would work, @lthls @mshinwell do you agree ?

Reply:
I don't like `Uses`, it's too generic. `Name_uses` would be ok, even though it tracks more than `Name.t`s, it's coherent with the various modules in `naming`.

Reply:
I agree that `Uses` is too generic.  However let's make sure the name reflects accurately what is going on, so probably not `Name_uses` either.

Reply:
Interrestingly, the information recorded in the module is enough to later compute a dominator (and even pdce I'd say) analysis, since we have the dependency graphs of all variables, as well as which variables are used and bound in which continuation, and which continuation calls which other one. I don't know if that helps with finding a good module name though.

Reply:
@chambart  suggests `Data_flow_graph`, or `Data_flow`, based on the explanation above.